packages/xo-web/src/xo-app/host/tab-patches.js

```diff
@@ -118,10 +189,84 @@ const INSTALLED_PATCH_COLUMNS_2 = [
   },
 ]
 
+class XcpPatches extends Component {
+  static contextTypes = {
+    router: React.PropTypes.object,
+  }
+
+  _chooseActionPatch = async doInstall => {
+    const choice = await chooseAction({
+      body: <p>{_('installPatchWarningContent')}</p>,
+      buttons: [
+        {
+          label: _('installPatchWarningResolve'),
+          value: 'install',
+          btnStyle: 'primary',
+        },
+        { label: _('installPatchWarningReject'), value: 'goToPool' },
+      ],
+      title: _('installPatchWarningTitle'),
+    })
+
+    return choice === 'install'
+      ? doInstall()
+      : this.context.router.push(`/pools/${this.props.host.$pool}/patches`)
```

Comment:
Why not using the `redirectUrl` param of the button?

Reply:
The modal button? We're not using `ActionButton`s there.

Reply:
Hmm, you're right, my bad.